instrumentation/opentelemetry-instrumentation-tornado/tests/test_instrumentation.py

```diff
@@ -602,6 +603,46 @@ def client_response_hook(span, response):
         self.memory_exporter.clear()
 
 
+class TestTornadoHTTPClientInstrumentation(TornadoTest, WsgiTestBase):
+    def test_http_client_success_response(self):
+        response = self.fetch("/")
+        self.assertEqual(response.code, 201)
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 3)
+        manual, server, client = self.sorted_spans(spans)
+        self.assertEqual(manual.name, "manual")
+        self.assertEqual(server.name, "GET /")
+        self.assertEqual(client.name, "GET")
+        self.memory_exporter.clear()
+
+    def test_http_client_failed_response(self):
+        # when an exception isn't thrown
+        response = self.fetch("/some-404")
+        self.assertEqual(response.code, 404)
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 2)
+        server, client = self.sorted_spans(spans)
+        self.assertEqual(server.name, "GET /some-404")
+        self.assertEqual(client.name, "GET")
+        self.memory_exporter.clear()
+
+        # when an exception is thrown
```

Comment:
nit: Does it make sense to add `self.assertEqual(len(memory_exporter.get_finished_spans()), 0)` here and maybe at the beginning of each test?

Reply:
Sounds like a worthy improvement to avoid test breakage due to leakage from forgetting to do cleanup in tests.

How about adding it to `TornadoTest.setUp()` instead of into each test case?

Reply:
There's quite a few tests that do multiple rounds of instrumentation, requiring manually clearing the exporter as test steps. Given this, I think it's actually better as is for consistency.

Reply:
I'll look into it

Reply:
I tried to move exporter clearing to `setUp()`, and remove redundant clears from tests. I couldn't get it working nicely, after removing what I thought were redundant cases there was test failures.

I'm not willing to spend more time working on this (I'm short on time).